test/unit/formatting_stack/impl/overrides.clj

```diff
@@ -0,0 +1,97 @@
+(ns unit.formatting-stack.impl.overrides
+  (:require
+   [clojure.test :refer [are deftest is testing]]
+   [formatting-stack.impl.overrides :as sut]))
+
+(deftest removing-deep-merge
+  (are [a b expected] (testing (pr-str [a b])
+                        (is (= expected
+                               (sut/removing-deep-merge a b)))
+                        true)
+    {}               nil                          nil
+    nil              {}                           {}
+    {}               {:a 2}                       {:a 2}
+    {:a 2}           {}                           {:a 2}
+    {:a 2}           {:b 3}                       {:a 2 :b 3}
+    {:a 2}           {:a nil}                     {}
+    {:a {:b {:c 1}}} {:a {:b {:c 2}}}             {:a {:b {:c 2}}}
+    {:a {:b {:c 1}}} {:a {:b {:c 2 :d 3} :e 4}}   {:a {:b {:c 2 :d 3} :e 4}}
+    {:a {:b {:c 1}}} {:a {:b {:c nil}}}           {:a {:b {}}}
+    {:a {:b {:c 1}}} {:a {:b {:c nil :d {:e 2}}}} {:a {:b {:d {:e 2}}}}))
+
+(deftest apply-overrides
+  (let [sample-members [{:id :a/id :age 42 :bar {:baz 42}}]
+        reified-member (reify)]
+    (are [desc members overrides expected] (testing (pr-str [desc members overrides])
+                                             (is (= expected
+                                                    (try
+                                                      (sut/apply-overrides members overrides)
+                                                      (catch AssertionError _
+                                                        ::EXCEPTION))))
```

Comment:
Why not rethrow this exception? Currently it'll swallow the error.

Reply:
There's an `are` test case that will expect this `::EXCEPTION`, so these aren't omitted

(that's part of why I put these in caps - to make it more immediately visible that something is both declared and consumed)

Reply:
Could we split that `are` up, so the ones expecting an exception can use `thrown-with-msg?` for a more specific test? (e.g. now _any_ assertionerror will pass)

Reply:
Maybe I wouldn't split the `are`- a single `are` means more contrast, less apples-vs-oranges etc

Will look into making this `are` itself more accurate, should be possible